runtime/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/config/MqttBindingConfig.java

```diff
@@ -128,13 +134,47 @@ public ModelConfig supplyModelConfig(
         ModelConfig config = null;
         if (topics != null)
         {
-            for (Map.Entry<Matcher, ModelConfig> t : topics.entrySet())
+            for (Map.Entry<Matcher, TopicValidator> t : topics.entrySet())
             {
                 final Matcher matcher = t.getKey();
                 matcher.reset(topic);
                 if (matcher.find())
                 {
-                    config = t.getValue();
+                    config = t.getValue().content;
+                    break;
+                }
+            }
+        }
+        return config;
+    }
+
+    public ModelConfig supplyUserPropertyModelConfig(
+        String topic,
+        String16FW userPropertyKey)
+    {
+        ModelConfig config = null;
+        if (topics != null)
+        {
+            for (Map.Entry<Matcher, TopicValidator> t : topics.entrySet())
+            {
+                final Matcher matcher = t.getKey();
+                matcher.reset(topic);
+                if (matcher.find())
+                {
+                    Map<Matcher, ModelConfig> userProperties = t.getValue().userProperties;
+                    if (userProperties != null)
+                    {
+                        for (Map.Entry<Matcher, ModelConfig> u : userProperties.entrySet())
+                        {
+                            final Matcher userPropertyMatcher = u.getKey();
+                            userPropertyMatcher.reset(userPropertyKey.asString());
+                            if (userPropertyMatcher.find())
+                            {
+                                config = u.getValue();
+                                break;
+                            }
+                        }
+                    }
```

Comment:
Any reason we need to iterator over matchers instead of looking up user property by name directly using string key in the map to find the right model?

Also, perhaps we can pre-create the map with `String16FW` keys so we can lookup the model for validation without creating the string dynamically each time from the `String16FW` passed in?